aws_lambda_powertools/utilities/data_classes/kinesis_firehose_event.py

```diff
@@ -77,6 +196,26 @@ def data_as_json(self) -> dict:
             self._json_data = self._json_deserializer(self.data_as_text)
         return self._json_data
 
+    def create_firehose_response_record(
```

Comment:
```suggestion
    def build_data_transformation_response(
```

tip: think **Intent** first, then align what's the documentation is suggesting and how a customer eventually finds this information.

If we change the name now, you can easily find Kinesis Data Transformation docs as well as a few articles on how other customers handle data transformation in Firehose :)